synapse/storage/util/sequence.py

```diff
@@ -106,7 +106,15 @@ def __init__(self, sequence_name: str):
 
     def get_next_id_txn(self, txn: Cursor) -> int:
         txn.execute("SELECT nextval(?)", (self._sequence_name,))
-        return txn.fetchone()[0]
+        fetch_res = txn.fetchone()
+        if fetch_res is None:
+            # this should technically not happen, unless the connection/cursor/transaction has been corrupted
+            raise RuntimeError(
+                "nextval for self._sequence_name {} returned empty".format(
+                    repr(self._sequence_name)
+                )
+            )
```

Comment:
Since these cannot happen it probably makes more sense to do `assert fetch_res is not None`. (Same below.)